diff_drive_controller/src/diff_drive_controller.cpp

```diff
@@ -120,6 +120,76 @@ static bool getWheelRadius(
 
 namespace diff_drive_controller
 {
+  static void resize(trajectory_msgs::JointTrajectoryPoint& msg,
+      const size_t size)
+  {
+    msg.positions.resize(size);
+    msg.velocities.resize(size);
+    msg.accelerations.resize(size);
+    msg.effort.resize(size);
+  }
+
+  static void resize(DiffDriveControllerState& msg, const size_t size)
+  {
+    msg.joint_names.resize(size);
+
+    resize(msg.desired         , size);
+    resize(msg.actual          , size);
+    resize(msg.limited         , size);
+    resize(msg.error           , size);
+    resize(msg.actual_estimated, size);
+    resize(msg.error_estimated , size);
+
+    resize(msg.actual_side_average          , 2);
+    resize(msg.error_side_average           , 2);
+    resize(msg.actual_estimated_side_average, 2);
+    resize(msg.error_estimated_side_average , 2);
+  }
+
+  static void error(
+      std::vector<double>& err,
+      const std::vector<double>& desired,
+      const std::vector<double>& actual)
+  {
+    const size_t n = err.size();
+
+    ROS_ASSERT(n == desired.size() && n == actual.size());
+
+    for (size_t i = 0; i < n; ++i)
+    {
+      err[i] = desired[i] - actual[i];
+    }
```

Comment:
You could do instead:
` std::transform (desired.begin(), desired.end(), actual.begin(), err.begin(), std::minus<double>());`